src/cli/restore/onedrive_test.go

```diff
@@ -50,3 +50,32 @@ func (suite *OneDriveSuite) TestAddOneDriveCommands() {
 		})
 	}
 }
+
+func (suite *OneDriveUnitSuite) TestAddOneDriveCommands_hasRestoreFlag() {
+	expectUse := oneDriveServiceCommand + " " + oneDriveServiceCommandUseSuffix
+
+	table := []struct {
+		name        string
+		use         string
+		expectUse   string
+		expectShort string
+		expectRunE  func(*cobra.Command, []string) error
+	}{
+		{"restore onedrive", restoreCommand, expectUse, oneDriveRestoreCmd().Short, restoreOneDriveCmd},
+	}
+	for _, test := range table {
+		suite.Run(test.name, func() {
+			t := suite.T()
+
+			cmd := &cobra.Command{Use: test.use}
+
+			c := addOneDriveCommands(cmd)
+			require.NotNil(t, c)
+
+			cmd.SetArgs([]string{"onedrive", "--restore-permissions"})
+
+			err := cmd.Execute()
+			assert.NotEqual(t, err.Error(), `unknown flag: --restore-permissions`)
```

Comment:
Not sure if this test is too hacky. I don't know if there is a simpler way to test this with our current setup.

Reply:
I don't think you need to execute the command.  Since flags are stored in the command, once you call `addOneDriveCommands(cmd)`, you can get check for the flag from the command with something like https://pkg.go.dev/github.com/spf13/cobra#Command.Flag.  Or you could iterate through https://pkg.go.dev/github.com/spf13/cobra#Command.LocalFlags.

Reply:
Yeah, having tests for flags seems like simple enough, but pretty useful. Let me [add](https://github.com/alcionai/corso/issues/2975) a `tech-debt` issue for now.